include/podio/Reader.h

```
@@ -73,50 +83,139 @@ class Reader {
     return getEntries(podio::Category::Event);
   }
+
+  /// Get the build version of podio that has been used to write the current
+  /// file
+  ///
+  /// @returns The podio build version
   podio::version::Version currentFileVersion() const {
     return m_self->currentFileVersion();
   }
+
+  /// Get the names of all the available Frame categories in the current file(s).
+  ///
+  /// @returns The names of the available categories from the file
   std::vector<std::string_view> getAvailableCategories() const {
     return m_self->getAvailableCategories();
   }
+
+  /// Get the datamodel definition for the given name
+  ///
+  /// @param name The name of the datamodel
+  ///
+  /// @returns The high level definition of the datamodel in JSON format
   const std::string_view getDatamodelDefinition(const std::string& name) const {
     return m_self->getDatamodelDefinition(name);
   }
+
+  /// Get all names of the datamodels that are available from this reader
+  ///
+  /// @returns The names of the datamodels
   std::vector<std::string> getAvailableDatamodels() const {
     return m_self->getAvailableDatamodels();
   }
 };
 
+/// Create a Reader is able to read the file
+///
+/// This will inspect the filename as well as peek at the file contents to
+/// instantiate the correct low level reader to open and read the file
+///
+/// @param filename The (path to the) file to read from
+///
+/// @returns A Reader that has been initialized and that can be used for reading
+///          data from the passed file
 Reader makeReader(const std::string& filename);
+
+/// Create a Reader that is able to read the files
+///
+/// This will inspect the filenames as well as peek into the **first file only**
+/// to decide based on the contents which low level reader to instantiate for
+/// reading. All files are assumed to be of the same I/O format, no switching
+/// between formats is possible.
+///
+/// @note For SIO files this will only read the first file!
```

we should put in some exception handling or something of that kind. That is too burried in the docs.

---

You mean, throw an exception when more than one SIO file is passed?

---

we should not silently ignore faulty states.

---

so yes